elyra/pipeline/airflow/provider_package_catalog_connector/airflow_provider_package_catalog_connector.py

```diff
@@ -0,0 +1,323 @@
+#
+# Copyright 2018-2022 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import ast
+import os
+from os.path import sep
+from pathlib import Path
+import re
+import shutil
+from tempfile import mkdtemp
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from urllib.parse import urlparse
+import zipfile
+
+import requests
+
+from elyra.pipeline.catalog_connector import ComponentCatalogConnector
+
+
+class AirflowProviderPackageCatalogConnector(ComponentCatalogConnector):
+    """
+    Read component definitions from an Apache Airflow provider package archive
+    """
+
+    REQUEST_TIMEOUT = 30
+
+    def get_catalog_entries(self, catalog_metadata: Dict[str, Any]) -> List[Dict[str, Any]]:
+
+        """
+        Returns a list of component_metadata instances, one per component found in the given registry.
```

Comment:
I think in #2347 this docstring is more specific to the connector being implemented. Wasn't sure if we wanted to do that here as well

Reply:
Yup! Will update. Since this connector is meant to be a template, it's better to be specific.